full-service/src/db/view_only_account.rs

```diff
@@ -0,0 +1,288 @@
+
+    /// Update an view-only-account name.
+    /// The only updatable field is the name. Any other desired update requires
+    /// adding a new account, and deleting the existing if desired.
+    fn update_name(
+        &self,
+        new_name: &str,
+        conn: &PooledConnection<ConnectionManager<SqliteConnection>>,
+    ) -> Result<(), WalletDbError>;
+
+    /// Update the next block index this account will need to sync.
+    fn update_next_block_index(
+        &self,
+        next_block_index: i64,
+        conn: &PooledConnection<ConnectionManager<SqliteConnection>>,
+    ) -> Result<(), WalletDbError>;
+
+    /// Delete a view-only-account.
+    fn delete(
+        self,
+        conn: &PooledConnection<ConnectionManager<SqliteConnection>>,
+    ) -> Result<(), WalletDbError>;
+}
+
+impl ViewOnlyAccountModel for ViewOnlyAccount {
+    fn create(
+        account_id_hex: &str,
+        view_private_key: &RistrettoPrivate,
+        first_block_index: i64,
+        import_block_index: i64,
+        name: &str,
+        conn: &PooledConnection<ConnectionManager<SqliteConnection>>,
+    ) -> Result<ViewOnlyAccount, WalletDbError> {
+        use crate::db::schema::view_only_accounts;
+
+        let encoded_key = mc_util_serial::encode(view_private_key);
+
+        let new_view_only_account = NewViewOnlyAccount {
+            account_id_hex,
+            view_private_key: &encoded_key,
+            first_block_index,
+            // next block index will always be the same as
+            // first block index when importing an account
+            next_block_index: first_block_index,
+            import_block_index,
+            name,
+        };
+
+        diesel::insert_into(view_only_accounts::table)
+            .values(&new_view_only_account)
+            .execute(conn)?;
+
+        ViewOnlyAccount::get(account_id_hex, conn)
+    }
+
+    fn get(
+        account_id: &str,
+        conn: &PooledConnection<ConnectionManager<SqliteConnection>>,
+    ) -> Result<ViewOnlyAccount, WalletDbError> {
+        use crate::db::schema::view_only_accounts::dsl::{
+            account_id_hex as dsl_account_id, view_only_accounts,
+        };
+
+        match view_only_accounts
+            .filter((dsl_account_id).eq(account_id))
+            .get_result::<ViewOnlyAccount>(conn)
+        {
+            Ok(a) => Ok(a),
+            // Match on NotFound to get a more informative NotFound Error
+            Err(diesel::result::Error::NotFound) => Err(WalletDbError::AccountNotFound(
+                // TODO(cc) improve this error handling
+                // str::from_utf8(view_private_key)?,
+                "account not found".to_string(),
+            )),
+            Err(e) => Err(e.into()),
+        }
+    }
+
+    fn list_all(
+        conn: &PooledConnection<ConnectionManager<SqliteConnection>>,
+    ) -> Result<Vec<ViewOnlyAccount>, WalletDbError> {
+        use crate::db::schema::view_only_accounts;
```

Comment:
Optional/Equivalent: Add a top-level `use crate::db::schema` then these in-method declarations can be `use schema::...`

Reply:
this makes sense, but do you know why Diesel suggests inlining this import at all? I don't see it mentioned in the documentation and it seems to work when not inlined.